CHANGES.md

```diff
@@ -1,3 +1,58 @@
+## Version 0.14.0
+
+This release is a major milestone for Spire.
+
+First of all, it introduces a dependency on typelevel/algebra (and
+thus on typelevel/cats-kernel). This provides immediate compatibility
+with Cats and Algebird.
+
+Second of all, Spire is now published to the org.typelevel
+organization (which was previously org.spire-math).
+
+Additions:
+ * Add GCDRing, generalized from EuclideanRing
+ * Add DivisionRing, a field without commutative multiplication
+
+Changes:
+ * Many type classes are now aliases to algebra's type classes
+ * The relationship between EuclideanRing and Gcd has changed
```

Comment:
We should also mention that GCDRing and EuclideanRing operations (gcd, lcm, quot, mod, quotmot) have been made lawful for all types.

We now require `gcd` and `lcm` commutative.

For integers, the result of the above operations is unchanged (apart fixing the corner cases of gcd/lcm).

But Rational, Double, Float... implemented originally EuclideanRing with truncated division, which is incorrect and will affect e.g. polynomial rings as well. This has been corrected. In some cases, it impacts gcd/lcm as well.

Trouble now: Java/Scala do not represent Euclidean division for `/~` `%` and `/%`, rather truncated division. So our syntaxes diverge there (this affects only stdlib Float/Double, which is why the issue never popped up in tests).